ryu/d2s.c

```diff
@@ -49,18 +49,84 @@
 #define HAS_64_BIT_INTRINSICS
 #endif
 
+#if defined(_M_IX86)
+#define RYU_DIVCONST 1
```

Comment:
Can we detect when this is necessary and automatically enable it in those cases? Also, I'd like to see an upstream bug filed and referenced here. I hereby declare my intent to delete this code 5 years after the upstream bug(s) are fixed. Finally, it would also be good if we could get test coverage on CI for this.

Reply:
I'm not sure if we can detect automatically if this is necessary.

However, I can fully understand that you want to keep the implementation clean. We could implement some "intrinsics" like `div10` in a separate file and update the code to use these intrinsics. This keeps the code reasonably clean while providing the opportunity to optimize in case it matters.

Would you be ok with something like this:
https://github.com/ulfjack/ryu/compare/master...abolz:intrinsics

Tests definitely must be added. (On my test PC, the performance for MSVC x64 is unaffected.)

Reply:
I like the intrinsics branch better than this one.

Reply:
I'm not going to leave out a 40-50% improvement even if it's only 32-bit, and makes the code a little less readable.